common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/core/JavaSTFileSwitch.java

```diff
@@ -0,0 +1,87 @@
+package io.vrap.rmf.codegen.common.generator.core;
+
+import io.vrap.rmf.codegen.common.processor.extension.ExtensionMapper;
+import io.vrap.rmf.raml.model.resources.Resource;
+import io.vrap.rmf.raml.model.types.ObjectType;
+import io.vrap.rmf.raml.model.types.StringType;
+import io.vrap.rmf.raml.model.types.util.TypesSwitch;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.ComposedSwitch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.stringtemplate.v4.STGroupFile;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class JavaSTFileSwitch extends ComposedSwitch<STGroupFile> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JavaSTFileSwitch.class);
+
+    private Map<String, STGroupFile> cache = new HashMap<>();
```

Comment:
I'm not 100% sure, but I thought that stringtemplate already provides a cache. But let's keep your code as it is and we can still see if we can improve this further in the future.